### PR TITLE
expression: ignore `TypeNULL` in `HandleBinaryLiteral`

### LINKED ISSUE #49526: `HandleBinaryLiteral` should ignore `TypeNull`
## Enhancement

`from_binary(<nil>)` actually does nothing, but TiDB regards the `NULL` as binary string, so it may avoid some expressions to be pushed down to TiFlash. We can manually ignore this case.

(Though I don't know why this constant is not folded.)

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Issue Number: close #49526

Problem Summary:

### What changed and how does it work?

Ignore the `TypeNull` in `HandleBinaryLiteral`, as the `castStringToString` will actually do nothing for NULL (except changing the type).

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: tests/integrationtest/r/expression/cast.result [hunk 1]
```diff
@@ -97,3 +97,10 @@ select 1 from t where y = YEAR(CURDATE());
 select cast(cast('14:15' as time) as year) = YEAR(CURDATE());
 cast(cast('14:15' as time) as year) = YEAR(CURDATE())
 1
+explain select null as a union all select 'a' as a;
```
Comment (windtalker) [inline tests/integrationtest/r/expression/cast.result:100]: What is the explain result without this pr?

Comment (YangKeao) [reply tests/integrationtest/r/expression/cast.result:100]:
```
mysql> explain select null as a union all select 'a' as a;
+------------------------+---------+------+---------------+---------------------------------------------------+
| id                     | estRows | task | access object | operator info                                     |
+------------------------+---------+------+---------------+---------------------------------------------------+
| Union_8                | 2.00    | root |               |                                                   |
| ├─Projection_10        | 1.00    | root |               | cast(from_binary(<nil>), var_string(1))->Column#3 |
| │ └─TableDual_11       | 1.00    | root |               | rows:1                                            |
| └─Projection_12        | 1.00    | root |               | a->Column#3                                       |
|   └─TableDual_13       | 1.00    | root |               | rows:1                                            |
+------------------------+---------+------+---------------+---------------------------------------------------+
5 rows in set (0.00 sec)
```

**Review by windtalker — APPROVED** [review]
LGTM

**Review by jackysp — APPROVED** [review]

**Pull request merged**